nominal/core/client.py

```diff
@@ -511,30 +514,48 @@ def create_dataset_from_io(
         )
-        request = ingest_api.IngestRequest(
-            options=ingest_api.IngestOptions(
+
+        source = ingest_api.IngestSource(s3=ingest_api.S3IngestSource(path=s3_path))
+        target = ingest_api.DatasetIngestTarget(
+            new=ingest_api.NewDatasetIngestDestination(
+                labels=list(labels),
+                properties={} if properties is None else dict(properties),
+                channel_config=_build_channel_config(prefix_tree_delimiter),
+                dataset_description=description,
+                dataset_name=name,
+                workspace=self._clients.workspace_rid,
+            )
+        )
+        timestamp_metadata = ingest_api.TimestampMetadata(
+            series_name=timestamp_column,
+            timestamp_type=_to_typed_timestamp_type(timestamp_type)._to_conjure_ingest_api(),
+        )
+
+        if file_type.is_parquet():
+            options = ingest_api.IngestOptions(
+                parquet=ingest_api.ParquetOpts(
+                    source=source,
+                    target=target,
+                    timestamp_metadata=timestamp_metadata,
+                    channel_prefix=channel_prefix,
+                    tag_columns=dict(tag_columns) if tag_columns else None,
+                    is_archive=file_type.is_parquet_archive(),
+                )
+            )
+        elif file_type.is_csv():
+            options = ingest_api.IngestOptions(
                 csv=ingest_api.CsvOpts(
-                    source=ingest_api.IngestSource(s3=ingest_api.S3IngestSource(path=s3_path)),
-                    target=ingest_api.DatasetIngestTarget(
-                        new=ingest_api.NewDatasetIngestDestination(
-                            labels=list(labels),
-                            properties={} if properties is None else dict(properties),
-                            channel_config=_build_channel_config(prefix_tree_delimiter),
-                            dataset_description=description,
-                            dataset_name=name,
-                            workspace=self._clients.workspace_rid,
-                        )
-                    ),
-                    timestamp_metadata=ingest_api.TimestampMetadata(
-                        series_name=timestamp_column,
-                        timestamp_type=_to_typed_timestamp_type(timestamp_type)._to_conjure_ingest_api(),
-                    ),
-                    additional_file_tags=None,
+                    source=source,
+                    target=target,
+                    timestamp_metadata=timestamp_metadata,
                     channel_prefix=channel_prefix,
-                    tag_keys_from_columns=None,
+                    tag_columns=dict(tag_columns) if tag_columns else None,
                 )
             )
-        )
+        else:
+            raise ValueError(f"Expected filetype {file_type} to be parquet or csv!")
```

Comment:
this is likely to be a breaking change for any script that passed in their own tuple (of which I've written a few, and probably others have as well)